src/net.cpp

```diff
@@ -235,10 +235,10 @@ bool IsPeerAddrLocalGood(CNode *pnode)
 
 std::optional<CAddress> GetLocalAddrForPeer(CNode *pnode)
 {
-    CAddress addrLocal = GetLocalAddress(&pnode->addr, pnode->GetLocalServices());
+    CAddress addrLocal = GetLocalAddress(&pnode->addr);
     if (gArgs.GetBoolArg("-addrmantest", false)) {
         // use IPv4 loopback during addrmantest
-        addrLocal = CAddress(CService(LookupNumeric("127.0.0.1", GetListenPort())), pnode->GetLocalServices());
+        addrLocal = CAddress(CService(LookupNumeric("127.0.0.1", GetListenPort())), NODE_NONE);
```

Comment:
I think the line lower down:

```
addrLocal = CAddress{pnode->GetAddrLocal(), addrLocal.nServices, addrLocal.nTime};
```

No longer makes sense, since `addrLocal.nServices` will always be `NODE_NONE`. Perhaps change `addrLocal.nServices` to `NODE_NONE`, since it is the caller's job to set the services.

Perhaps even better would be to change `GetLocalAddrForPeer` and `GetLocalAddress` to return a `CService` instead of a `CAddress`, so the caller has to set the `nTime` and `nServices` fields (which might help avoid bugs like #25314).

Reply:
> Perhaps even better would be to change GetLocalAddrForPeer and GetLocalAddress to return a `CService`

Added an extra commit to do this. Makes it obvious that the caller is supposed to set the `nTime` and `nServices` fields.